### ISSUE TITLE
Fuel shuffling may duplicate in restart cases

### ISSUE THREAD

**Issue by keckler** [terrapower/armi#940]
When doing a restart case, one may specify the `startNode`, meaning the reactor may be loaded up at a non-BOC state.

Regardless, the operator still triggers the `interactBOC()` hook for the specified `startCycle`, presumably under the assumption that some things for the cycle will need to be initialized at BOC.
https://github.com/terrapower/armi/blob/5db113423d0981f8887809b3297b97e3117265e6/armi/operators/operator.py#L334-L347

This logic is not handled properly in the `interactBOC()` hook of the `FuelHandlerInterface`:
https://github.com/terrapower/armi/blob/5db113423d0981f8887809b3297b97e3117265e6/armi/physics/fuelCycle/fuelHandlerInterface.py#L56-L73

Basically, whatever is specified in the user's custom shuffle logic will be executed at the `interactBOC()` call, even if `startNode` is not at BOC. So if you restart from, say node 1 of a cycle, then the shuffle operations for that cycle will take place twice. This could result in a serious bug that would be very difficult to notice. For instance, its possible that you could restart the calculation at MOC but unknowingly put fresh fuel assemblies in the core, so they would be missing the burn history from the first half of the cycle.
